vault/logical_system.go

```diff
@@ -2473,6 +2478,54 @@ func (b *SystemBackend) responseWrappingUnwrap(ctx context.Context, te *logical.
 	return response, nil
 }
 
+func (b *SystemBackend) handleMetrics(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+
+	format := data.Get("format").(string)
+
+	acceptHeaders := req.Headers["Accept"]
+	if format == "prometheus" || (len(acceptHeaders) > 0 && strings.HasPrefix(acceptHeaders[0], "application/openmetrics-text")) {
+
+		metricsFamilies, err := prometheus.DefaultGatherer.Gather()
```

Comment:
I don't really see a reason any of this needs to sit in the vault package. Please create a `helpers/metricsutil` or similar package. It can expose a type that can wrap an InmemSink. Then it could have methods that can be called to get appropriate responses, e.g. `metricsObj.PrometheusResponse()` and `metricsObj.GenericResponse`.

Reply:
Ok, will do the review tomorrow
creating the object at config time, and passing it in the Core instead of inMemMetrics looks acceptable ?